### PR TITLE
Feat: Automated Local E2E Testing - Test 0 to 1

### PR DESCRIPTION
**Pull request by daniel-hutao**
Signed-off-by: Daniel Hu <tao.hu@merico.dev>

# Summary

Automated Local E2E Testing - Test 0 to 1

### CODE REVIEW

**Comment by daniel-hutao** [comment]
/cc @IronCore864 @lfbdev Since there are about 200 lines of changes in this pr after the smoke framework is implemented, I want to merge it first; the health check function of the remaining three plugins will be implemented in a separate commit.

Please take a review at this pr.

**Comment by daniel-hutao** [comment]
I deleted the smoke/argocd/argocd.go just now. already rebase & force pushed

**Review by lfbdev — APPROVED** [review]

File: internal/pkg/pluginengine/check.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package pluginengine
+
+import (
+	"log"
+
+	"github.com/merico-dev/stream/internal/pkg/configloader"
+)
+
+// CheckHealthy return true while all tools are healthy
```
Comment (IronCore864) [inline internal/pkg/pluginengine/check.go:9]: CheckHealthy return true while all tools are healthy ->

CheckHealthy return**s** true if all tools are healthy

**Review by IronCore864 — APPROVED** [review]

File: internal/pkg/pluginengine/check.go (hunk 1 above)
Comment (daniel-hutao) [reply internal/pkg/pluginengine/check.go:9]: get it

**Pull request merged**